plugins/inventory/kubevirt.py

```diff
@@ -453,230 +475,80 @@ def get_cluster_domain(self, client: K8SClient) -> Optional[str]:
             return None
         return obj.get("spec", {}).get("baseDomain")
 
-    def get_available_namespaces(self, client: K8SClient) -> List:
+    def get_resources(
+        self, client: K8SClient, api_version: str, kind: str, **kwargs
+    ) -> List[ResourceField]:
         """
-        get_available_namespaces lists all namespaces accessible with the
-        configured credentials and returns them.
+        get_resources uses a dynamic K8SClient to fetch resourches from the K8S API.
         """
-        v1_namespace = client.resources.get(api_version="v1", kind="Namespace")
+        client = client.resources.get(api_version=api_version, kind=kind)
         try:
-            obj = v1_namespace.get()
+            result = client.get(**kwargs)
         except DynamicApiError as exc:
             self.display.debug(exc)
             raise KubeVirtInventoryException(
-                f"Error fetching Namespace list: {self.format_dynamic_api_exc(exc)}"
+                f"Error fetching {kind} list: {self.format_dynamic_api_exc(exc)}"
             ) from exc
-        return [namespace.metadata.name for namespace in obj.items]
 
-    def get_vmis_for_namespace(
-        self, client: K8SClient, name: str, namespace: str, opts: GetVmiOptions
-    ) -> None:
+        return result.items
+
+    def get_available_namespaces(self, client: K8SClient) -> List[str]:
+        """
+        get_available_namespaces lists all namespaces accessible with the
+        configured credentials and returns them.
+        """
+        return [
+            namespace.metadata.name
+            for namespace in self.get_resources(client, "v1", "Namespace")
+        ]
+
+    def get_vms_for_namespace(
+        self, client: K8SClient, namespace: str, opts: InventoryOptions
+    ) -> List[ResourceField]:
         """
-        get_vmis_for_namespace lists all VirtualMachineInstances in a namespace
-        and adds groups and hosts to the inventory.
+        get_vms_for_namespace returns a list of all VirtualMachines in a namespace.
         """
-        vmi_client = client.resources.get(
-            api_version=opts.api_version, kind="VirtualMachineInstance"
+        return self.get_resources(
+            client,
+            opts.api_version,
+            "VirtualMachine",
+            namespace=namespace,
+            label_selector=opts.label_selector,
```

Comment:
Why do you pass the whole `opts` object if you just use `opts.label_selector`?

Reply:
`opts.api_version` is used too (see above). It think it is more convenient this way and it helps to keep the method signatures small.

Reply:
Oh, you right! Missed that line.